### PR TITLE
Extract `invoke.onDone` and `invoke.onError`

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: new-packages/ts-project/src/state.ts [hunk 1]
```diff
@@ -339,329 +322,323 @@ export function extractState(
...
-          if (!ts.isPropertyAssignment(prop)) {
-            ctx.errors.push({ type: 'state_property_unhandled' });
+        for (const transition of prop.initializer.properties) {
+          if (!ts.isPropertyAssignment(transition)) {
+            ctx.errors.push({ type: 'transition_property_unhandled' });
+            return;
+          }
+          const event = getPropertyKey(ctx, ts, transition);
+          if (!event) {
+            ctx.errors.push({ type: 'transition_property_unhandled' });
```
Comment (farskid) [inline new-packages/ts-project/src/state.ts:380]: nit: maybe we wan use `forEachStaticPropery` here too.

**Comment by farskid** [comment]
@Andarist onDone and onError actions are not extracted. Are they being added in a follow up?

**Review by farskid — APPROVED** [review]
Works great!

**Comment by Andarist** [comment]
> @Andarist onDone and onError actions are not extracted. Are they being added in a follow up?

They are extracted from what I can see - I have just pushed out an extra test case to be extra sure.

File: new-packages/ts-project/src/state.ts (hunk 1 above)
Comment (Andarist) [reply new-packages/ts-project/src/state.ts]: done

**Pull request merged**